Sources/GRPCCore/Transport/ServerTransport.swift

```diff
@@ -14,22 +14,27 @@
  * limitations under the License.
  */
 
+/// A protocol server transport implementations must conform to.
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
 public protocol ServerTransport: Sendable {
   typealias Inbound = RPCAsyncSequence<RPCRequestPart>
   typealias Outbound = RPCWriter<RPCResponsePart>.Closable
 
-  /// Starts the transport and returns a sequence of accepted streams to handle.
+  /// Starts the transport.
   ///
   /// Implementations will typically bind to a listening port when this function is called
-  /// and start accepting new connections. Each accepted inbound RPC stream should be published
-  /// to the async sequence returned by the function.
+  /// and start accepting new connections. Each accepted inbound RPC stream will be handed over to
+  /// the provided `streamHandler` to handle accordingly.
+  ///
+  /// Implementations should throw if the transport fails to start.
```

Comment:
I'm not convinced we need to make the closure throwing, we only call this method in `GRPCServer/run`, and in that case we don't need it to be throwing: we give the stream to the router to handle, and that method's non-throwing.

Reply:
I am confused about your response here since I didn't say we should make the closure throwing. This comment here is in line with what I commented above. I don't think just allowing throws to happen during start is sensible for transports.

Reply:
Ah you're right, I misunderstood your request. Okay, that actually makes sense to me. I'd like to double check with @glbrntt in case he had something else in mind here.

Reply:
> I think this is too limiting. They should throw when they encounter an error at any point.

I agree, but the docs don't say it can throw _only_ if the transport fails to start. I.e. the docs don't say that you can't throw errors in other situations.